openssl-dynamic/src/main/c/tcn.h

```diff
@@ -103,6 +104,14 @@ jstring         tcn_new_stringn(JNIEnv *, const char *, size_t);
 
 #define AJP_TO_JSTRING(V)   (*e)->NewStringUTF((e), (V))
 
+#define TCN_CHECK_NULL(V, M, R)                      \
+    TCN_BEGIN_MACRO                                  \
+        if (V == NULL) {                             \
+            tcn_ThrowIllegalArgumentException(e, M); \
```

Comment:
I wonder if we can use the `#` operator so that we don't need to require M at all:

    tcn_ThrowIllegalArgumentException(e, #V);

http://en.cppreference.com/w/cpp/preprocessor/replace#.23_and_.23.23_operators